src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -188,7 +205,11 @@ public synchronized void initHotReloading()
         List<HotReloadableFile> fileList = new ArrayList<>();
         if (maybeFileBasedPrivateKey && hasKeystore())
         {
-            fileList.add(new HotReloadableFile(keystore));
+            fileList.add(new HotReloadableFile(keystoreContext.keystore));
+        }
+        if (maybeFileBasedOutboundPrivateKey && hasKeystore())
```

Comment:
Should this be `hasOutboundKeystore()`?

Reply:
I have to recollect my memory about this but there was some challenge in having a Test work fine. May be it was not this particular code and something else but if @jyothsnakonisa modifies this as per this suggestion, we will assume all tests are run to ensure it is fine.